packages/internal/src/js-utils.js

```diff
@@ -87,3 +129,45 @@ export const makeMeasureSeconds = currentTimeMillisec => {
   };
   return measureSeconds;
 };
+
+/**
+ * Generalize Node.js util.promisify to support callbacks with multiple
+ * post-error parameters, which fulfill the returned promise as an array rather
+ * than as a single value.
```

Comment:
Post-error parameter? Maybe multiple parameters of which the last is an error?

Also what kind of node callback API does this!?

Reply:
> what kind of node callback API does this!?

The motivating use case: [`tmp.dir`](https://www.npmjs.com/package/tmp#asynchronous-directory-creation) expects a callback with parameters `(err, path, cleanupCallback)`.

But you were right to be unsettled by this function; I removed it and switched to the cleaner `const { name: dbDir, removeCallback: cleanupDB } = tmp.dirSync(…);`